diffrax/_step_size_controller/pid.py

```diff
@@ -545,6 +547,11 @@ def _scale(_y0, _y1_candidate, _y_error):
         if self.dtmin is not None:
             if not self.force_dtmin:
                 result = RESULTS.where(dt < self.dtmin, RESULTS.dt_min_reached, result)
+            # if we are already at dtmin and dt is unchanged (factor == 1),
+            # reset dt to dtmin to avoid accumulating float precision errors
+            dt = jnp.where(at_dtmin & (factor == 1), self.dtmin, dt)
```

Comment:
I think `dt = prev_dt * 1` should imply `dt = prev_dt` exactly regardless of precision errors / floating point weirdness. Did you find this line necessary?